<launch>

  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gazebo_gui" default="false"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>
  <arg name="verbose" default="false"/>
  <arg name="world_name" default="empty_world"/>

  <arg name="lidar_offset_x" default="0.0"/>
  <arg name="lidar_offset_y" default="0.0"/>
  <arg name="lidar_offset_z" default="0.0"/>
  <arg name="lidar_offset_roll" default="0.0"/>
  <arg name="lidar_offset_pitch" default="0.0"/>
  <arg name="lidar_offset_yaw" default="0.0"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch" >
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="gui" value="$(arg gazebo_gui)"/>
    <arg name="headless" value="$(arg headless)"/>
    <arg name="debug" value="$(arg debug)"/>
    <arg name="verbose" value="$(arg verbose)"/>
    <arg name="world_name" value="$(find nexplore_gz_simulator)/world/$(arg world_name).world"/>
  </include>

  <param name="lidar_description" command="$(find xacro)/xacro --inorder '$(find nexplore_gz_simulator)/urdf/lidar.urdf.xacro'" />
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_lidar" args="-urdf -param /lidar_description -model lidar -x 0 -y 0 -z 0.3"/>

  <node pkg="nexplore_gz_simulator" type="nexplore_gz_simulator" name="nexplore_gz_simulator" output="screen" required="true">
    <param name="lidar_offset_x" value="$(arg lidar_offset_x)" />
    <param name="lidar_offset_y" value="$(arg lidar_offset_y)" />
    <param name="lidar_offset_z" value="$(arg lidar_offset_z)" />
    <param name="lidar_offset_roll" value="$(arg lidar_offset_roll)" />
    <param name="lidar_offset_pitch" value="$(arg lidar_offset_pitch)" />
    <param name="lidar_offset_yaw" value="$(arg lidar_offset_yaw)" />
  </node>

</launch>
